src/nnbench/types/types.py

```diff
@@ -170,6 +172,24 @@ def __post_init__(self):
             super().__setattr__("name", self.fn.__name__)
         super().__setattr__("interface", Interface.from_callable(self.fn, self.params))
 
+        original_setUp = self.setUp
+
+        def wrapped_setUp(state: State, /, *args: Any, **kwargs: Any) -> None:
+            # TODO: setUp logic
+            print("SetUp: ", state)
+            original_setUp(*args, **kwargs)
+
+        super().__setattr__("setUp", wrapped_setUp)
+
+        original_tearDown = self.tearDown
+
+        def wrapped_tearDown(state: State, /, *args: Any, **kwargs: Any) -> None:
+            # TODO: tearDown logic
+            print("tearDown: ", state)
+            original_tearDown(*args, **kwargs)
+
+        super().__setattr__("tearDown", wrapped_tearDown)
+
```

Comment:
Ah, no - I meant we should change the setUp/tearDown calling convention, as in, require callables of signature `(state, params) => Any`, not wrap the callable itself.

(We will need to adjust the docs on setUp and tearDown tasks afterwards.)

NB: The `NoOp` from that module must then become `def NoOp(state, params): pass`.

I guess it is nicer to pass the dict instead of the kwargs. But this requires wrapping it into a form where the user cannot inject any parameters, e.g. `types.MappingProxy`.